src/avro_json_encoder.erl

```diff
@@ -32,9 +32,20 @@
 -export([encode_type/1]).
 -export([encode_value/1]).
 -export([encode_value/2]).
+-export([encode/3]).
 
+-include("erlavro.hrl").
 -include("avro_internal.hrl").
 
+-define(ENCODE_ARRAY_FUN(L),
+  fun(IType, Item) -> encode(L, IType, Item) end).
+-define(ENCODE_RECORD_FUN(L),
+  fun({FN, FT, FV}) -> {encode_string(FN), encode(L, FT, FV)} end).
+-define(ENCODE_MAP_FUN(L),
+  fun(IType, K, V) -> {encode_string(K), encode(L, IType, V)} end).
+-define(ENCODE_UNION_FUN(L),
+  fun(MemberT, Value, _) -> {encode_string(avro:get_type_fullname(MemberT)), encode(L, MemberT, Value)} end).
```

Comment:
i'd prefer having them where they are needed
it's easier to follow when reading code for the recursive `encode/3` call